core/ecschema-metadata/src/Deserialization/XmlSerializationUtils.ts

```diff
@@ -105,10 +102,11 @@ export namespace XmlSerializationUtils {
    */
   export async function writeStructProperty(propertyClass: StructProperty, propertyValue: any, structElement: Element, schemaDoc: Document): Promise<void> {
     const structClass = propertyClass.structClass;
-    if (!structClass.properties)
+    //TODO: should we be checking for any properties now instead of just local ones?
```

Comment:
can't we skip this as well?  now looping over the iterator returned by getPropertiesSync will do no more work than this check ... so no need to do the check